### ISSUE TITLE
Container Naming

### ISSUE THREAD

**Issue by vsoch** [wkharold/community#9]
Singularity allows you to specify a custom name for your container with `--name`, so it might make sense to allow users to do this with the Google Cloud Builder.

**Comment by wkharold** [comment]
Definitely. Is there a canonical naming scheme that we should use/promote?

**Comment by vsoch** [comment]
The traditional naming schemas all are relevant for Docker (err OCI) manifests, for example:

```
<registry>/<namespace>/<container>:<tag>@<digest>
```

but I don't know if this has a standard that maps to storage. For a bucket, it could still work like:

```
gs://<bucket>/<namespace>/<container>:<tag>@<digest>
```

but since we want Google Cloud Storage to parse "folders" in the names, perhaps the : and @ would be replaced with a slash:

```
gs://<bucket>/<namespace>/<container>/<tag>/<digest>
```

This is sort of along the lines of what I did for Singularity Hub, but it's definitely not a standard of any kind, just a "I think this is logical" method. I mentioned in the comment here:

![image](https://user-images.githubusercontent.com/814322/51778785-9e717480-20d1-11e9-9842-75c4cfbe7e94.png)

It might be good to reach out to OCI and ask if there is a standard for storage on the filesystem. If you look i Dockerlayer caches it's just a single list of sha256:xxxxxx tar.gz files... kind of messy! In that storage affords for human friendly browsing, this kind of cache isn't optimal.
